test/active_record_doctor/detectors/missing_presence_validation_test.rb

```diff
@@ -42,6 +42,16 @@ def test_non_null_column_is_not_reported_if_association_validation_present
     refute_problems
   end
 
+  def test_non_null_column_is_not_reported_if_polymorphic_association_validation_present
+    Context.create_table(:images) do |t|
+      t.references :imageable, null: false, polymorphic: true
+    end.define_model do
+      belongs_to :imageable, polymorphic: true, required: true
```

Comment:
Required is deprecated in newer versions of Rails. Let's add a conditional here and pass `required` for older versions use `optional` for newer versions.

Additionally, we should be testing for all combinations:

1. Columns are required, association is required (with covered this already)
2. Columns are required, association is optional.
3. Columns are optional, association is required.
4. Columns are optional, association is optional.